File: mt9x9.fs
(*
9x9 multiplication table in F#
CC0, Wei-Lun Chao <bluebat@member.fsf.org>, 2021.
*)
// fsharpc mt9x9.fs && mono mt9x9.exe || fsharpi --quiet --exec mt9x9.fs

module mt9x9
for i in 1..3..9 do
    for j in 1..9 do
        List.iter (fun k ->
            printf "%dx%d=%2d\t" k j (k*j)) [i; i+1; i+2]
        printf "\n"
    printfn ""

File: mt9x9.kt
/*
9x9 multiplication table in Kotlin
CC0, Wei-Lun Chao <bluebat@member.fsf.org>, 2022.
*/
// kotlinc-native -o mt9x9 mt9x9.kt && ./mt9x9.kexe

fun main() {
    for (i in 1..9 step 3) {
        for (j in 1..9) {
            (i..i+2).forEach {
                print("${it}x${j}=")
                print((it*j).toString().padStart(2) + "\t")
            }
            print("\n")
        }
        println()
    }
}
